### PR TITLE
[SDK] New version compatibility scheme

### PR DESCRIPTION
**Pull request by eanders-MS**
Soren and I worked out a new version checking scheme: Client reports current client version and minimum supported SDK version. App checks this against minimum supported client version and current SDK version. If current client version >= minimum supported client version && current SDK version >= minimum supported SDK version, the connection is allowed.

### CODE REVIEW

File: packages/sdk/src/utils/verifyClient.ts [hunk 1]
```diff
@@ -3,61 +3,83 @@
...
+
+/*
+ * Minimum Supported Client Library version
+ * As part of the manual SDK release procedures, this is reset to match CurrentSDKVersion Major.Minor if new functions
+ * have been added that don't work on older clients (i.e. pretty much every release). Since host apps are required to
+ * update client libraries regularly, this one is not a big deal to update.
  */
 // tslint:disable-next-line:variable-name
-const CurrentProtocolVersion = 1;
+const MinimumSupportedClientVersion = semver.coerce('0.5');
```
Comment (stevenvergenz) [inline packages/sdk/src/utils/verifyClient.ts]: Does this mean that client versions are expected to track their SDK versions?

File: packages/sdk/src/utils/verifyClient.ts [hunk 2]
```diff
@@ -3,61 +3,83 @@
...
+
+    // Test the current client version. Is it greater than or equal to the minimum client version?
+    const clientPass = semver.gte(CurrentClientVersion, MinimumSupportedClientVersion);
+    // Test the current SDK version. Is it greater than or equal to the minimum SDK version?
+    const sdkPass = semver.gte(CurrentSDKVersion, MinimumSupportedSDKVersion);
+
+    if (!clientPass) {
+        // tslint:disable-next-line:max-line-length
+        const message = `Connection rejected due to out of date client. Client version: ${CurrentClientVersion.toString()}. Min supported version by SDK: ${MinimumSupportedClientVersion.toString()}`;
+        log.info('network', message);
```
Comment (stevenvergenz) [inline packages/sdk/src/utils/verifyClient.ts]: >log.info('network' [](start = 8, length = 18)

I think these messages should be exposed to the app developer by default, not hidden with network traffic. Also, probably an `error` or `warning`, not an `info` message.

File: packages/sdk/src/utils/verifyClient.ts (hunk 1 above)
Comment (eanders-MS) [reply packages/sdk/src/utils/verifyClient.ts]: The client needs to track the minimum SDK version it supports, but it shouldn't change often. We must make efforts to remain as backward compatible as possible. Incrementing this version in the client is a Big Deal, and requires team discussion and signoff.

File: packages/sdk/src/utils/verifyClient.ts (hunk 2 above)
Comment (eanders-MS) [reply packages/sdk/src/utils/verifyClient.ts]: True, they shouldn't be in with the network traffic.

**Comment by sorenhan** [comment]
approved for RED branch

**Pull request merged**